edb/edgeql/parser/grammar/expressions.py

```diff
@@ -918,80 +917,16 @@ def reduce_NFCONST(self, *kids):
         self.val = qlast.DecimalConstant(value=kids[0].val)
 
 
-class RawStringConstant(Nonterm):
-
-    def reduce_RSCONST(self, str_tok):
-        match = lexutils.VALID_RAW_STRING_RE.match(str_tok.val)
-        if not match:
-            raise EdgeQLSyntaxError(
-                f"invalid raw string literal", context=str_tok.context)
-
-        quote = match.group('Q')
-        val = match.group('body')
-
-        self.val = qlast.RawStringConstant(value=val, quote=quote)
-
-
-class EscapedStringConstant(Nonterm):
-
-    def reduce_SCONST(self, str_tok):
-        match = lexutils.VALID_STRING_RE.match(str_tok.val)
-
-        if not match:
-            raise EdgeQLSyntaxError(
-                f"invalid string literal", context=str_tok.context)
-        if match.group('err_esc'):
-            raise EdgeQLSyntaxError(
-                f"invalid string literal: invalid escape sequence "
-                f"'{match.group('err_esc')}'",
-                context=str_tok.context)
-        elif match.group('err_cont'):
-            raise EdgeQLSyntaxError(
-                f"invalid string literal: invalid line continuation",
-                hint="newline has to immediately follow '\\'",
-                context=str_tok.context)
-
-        quote = match.group('Q')
-        val = match.group('body')
-
-        # collapse the whitespace after a line continuation
-        val = lexutils.collapse_newline_whitespace(val)
-
-        self.val = qlast.StringConstant(value=val, quote=quote)
-
-
 class BaseStringConstant(Nonterm):
 
-    def reduce_EscapedStringConstant(self, *kids):
-        self.val = kids[0].val
-
-    def reduce_RawStringConstant(self, *kids):
-        self.val = kids[0].val
+    def reduce_SCONST(self, token):
+        self.val = qlast.RawStringConstant(value=token.clean_value)
```

Comment:
If we're only using one AST node in the parser it should be `StringConstant` rather than `RawStringConstant`. In general, it appears that we should then remove the `RawStringConstant` altogether having only one type of AST node.

Reply:
Done